storage/innobase/log/log0log.cc

```diff
@@ -522,7 +524,7 @@ log_t::resize_start_status log_t::resize_start(os_offset_t size) noexcept
         {
           memcpy_aligned<16>(resize_buf, buf, (buf_free + 15) & ~15);
           start_lsn= first_lsn +
-            (~lsn_t{get_block_size() - 1} & (write_lsn - first_lsn));
+            (~lsn_t{write_size - 1} & (write_lsn - first_lsn));
```

Comment:
I thought that it continues to make sense to use `get_block_size()` when reading the two checkpoint blocks in `recv_sys_t::find_checkpoint()`. OK, it would also make sense to read the first 0x3000 bytes in a single request and then scan the memory buffer for the log header and checkpoints, in whatever format they are in. But I do not oppose; we might as well use `log_sys.write_size` for now. Come to think about it, we can remove the conditional members `log_sys.block_size` and `log_t::get_block_size()` altogether.

For writing a checkpoint, you are right that we’d better use `log_sys.write_size` (as long as it is between 64 and 4096 bytes). I will fix that. The remaining references to `get_block_size()` are/were in debug assertions.

Reply:
4ca355d863e3b6a14439eebbb2958afccb3548e3 basically replaces `log_sys.block_size` with `log_sys.write_size`. It will also use `innodb_log_write_ahead_size` in `mariadb-backup --backup`.